### PR TITLE
Extend fuzzing to other map types

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Signed-off-by: Alan Jowett <alanjo@microsoft.com>

## Description

Add some other map types to the fuzzing test.

## Testing

New test.

## Documentation

N/A.

Partially resolves: #979

### CODE REVIEW

File: libs/execution_context/ebpf_maps.c [hunk 1]
```diff
@@ -746,6 +746,10 @@ _delete_hash_map(_In_ _Post_invalid_ ebpf_core_map_t* map)
 static void
 _delete_object_hash_map(_In_ _Post_invalid_ ebpf_core_map_t* map)
 {
+    if (!map) {
+        return;
+    }
```
Comment (dthaler) [inline libs/execution_context/ebpf_maps.c:751]: Why is this needed?  Since the SAL annotation has _In_ it should be the caller's responsibility to check this.
If this code path could ever be hit, it means that the caller should have already seen an error due to the SAL annotation.

File: tests/fuzz/execution_context.cpp [hunk 2]
```diff
@@ -24,38 +24,52 @@
...
     fd_t program_fd;
+    std::vector<std::string> map_names{
+        "HASH",
+        "PERCPU_HASH",
+        "ARRAY",
+        "PERCPU_ARRAY",
+        "LRU_HASH",
+        "LRU_PERCPU_HASH",
+        "QUEUE",
+        "STACK",
```
Comment (dthaler) [inline tests/fuzz/execution_context.cpp:46]: Is there some logical order to this list?  It doesn't seem to be in alphabetic or numerical order.

File: tests/fuzz/execution_context.cpp [hunk 3]
```diff
@@ -118,6 +132,16 @@ TEST_CASE("execution_context_direct", "[fuzz]")
     ebpf_fuzzing_enabled = true;
     auto mt = seed_random_engine();
 
+    // Limit this processes memory to 50Mb
```
Comment (dthaler) [inline tests/fuzz/execution_context.cpp:135]:
```suggestion
    // Limit this processes memory to 50Mb.
```
But did you really mean "MB" (megabytes) instead of "Mb" (megabits)?  Line 139 suggests you meant "MB"

File: libs/execution_context/ebpf_maps.c (hunk 1 above)
Comment (Alan-Jowett) [reply libs/execution_context/ebpf_maps.c:751]: Failed map allocation triggers an AV here.

File: tests/fuzz/execution_context.cpp (hunk 2 above)
Comment (Alan-Jowett) [reply tests/fuzz/execution_context.cpp:46]: It's the same order they occur in map.o ELF file.

File: libs/execution_context/ebpf_maps.c (hunk 1 above)
Comment (dthaler) [reply libs/execution_context/ebpf_maps.c:750]: I don't doubt that. But as is, this is not the correct fix.  Line 747 says the contract is that the caller's responsibility is to never call it with null.  So unless you change the contract for all delete_entry() functions, then the check here needs to be moved out to the caller so that this function is never invoked with a null.

File: tests/fuzz/execution_context.cpp (hunk 2 above)
Comment (dthaler) [reply tests/fuzz/execution_context.cpp:46]: Ok.  it's unfortunate that map.o uses a random ordering.  Does the order here need to match the order in the map.o file?  If so, at least add a // comment explaining that.

**Review by dthaler — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]
